### PR TITLE
Document Device Trust App Access support

### PR DESCRIPTION
**Pull request by codingllama**
Document App Access device enforcement via roles and `tsh proxy app`.

I've added an App Access entry to the troubleshooting section, as referenced by #35274 / [lib/srv/app/server.go#L811][1].

I've also downgraded the "warning" sections to "notice" - the feature gap for device trust closes significantly in v15 with Linux and App Access (proxied), so that seemed fair.

https://github.com/gravitational/teleport.e/issues/2121

[1]: https://github.com/gravitational/teleport/blob/ed748287037dc51bd58bc2e9ca8b6eda9dbb135c/lib/srv/app/server.go#L811

### CODE REVIEW

File: docs/pages/access-controls/device-trust/enforcing-device-trust.mdx [hunk 1]
```diff
@@ -148,6 +152,56 @@ cluster. Likewise, a root cluster will not enforce device trust on resources in
...
+For example, to enforce device trust for all `env:production` apps, save the
+role below as `require-trusted-device-apps.yaml`:
+
+```yaml
+kind: role
+version: v7
+metadata:
+  name: require-trusted-device-apps
+spec:
+  options:
```
Comment (zmb3) [inline docs/pages/access-controls/device-trust/enforcing-device-trust.mdx:173]: How did you end up doing this? Most of our role options apply unconditionally to anyone who has the role, but it looks like in this case the `device_trust_mode` option only applies when there's a match on `app_labels`?

Comment (codingllama) [reply docs/pages/access-controls/device-trust/enforcing-device-trust.mdx:173]: I stole it from require_session_mfa, which also only applies to the labels. :)

Comment (codingllama) [reply docs/pages/access-controls/device-trust/enforcing-device-trust.mdx:173]: You can follow the trail from here, if you are curious: https://github.com/gravitational/teleport/blob/cc7608b9da600a7f9da4fc5ce2113b6aafd44653/lib/services/role.go#L2426.

Comment (codingllama) [reply docs/pages/access-controls/device-trust/enforcing-device-trust.mdx:173]: The device trust role option always works like that, it's mentioned in the role-based section: https://github.com/gravitational/teleport/pull/35335/files#diff-50ffd70881a0efbcd6a76407915f5e65f2b4bf746e3c953991e453b8b2bbdb9fR39-R42

I'll take a look at the roles page, keeping this open to remind me.

**Review by zmb3 — APPROVED** [review]

**Comment by codingllama** [comment]
Friendly ping @ptgott ?

File: docs/pages/access-controls/device-trust.mdx [hunk 2]
```diff
@@ -5,14 +5,14 @@ layout: tocless-doc
...
+<Admonition type="notice">
   Device Trust supports the following components:
 
   - User devices: macOS, Windows and Linux.
   - Teleport client: `tsh` and Teleport connect.
-  - Resources: SSH, Database and Kubernetes.
+  - Resources: Apps (role-based only), SSH, Database and Kubernetes
 
-  Support for Web UI and application access are planned for upcoming Teleport
+  Support for Web UI and Desktop Access are planned for upcoming Teleport
```
Comment (ptgott) [inline docs/pages/access-controls/device-trust.mdx:15]:
```suggestion
  Support for the Web UI and desktop access are planned for upcoming Teleport
```

We're no longer dividing Teleport into separate, resource-based products, so we don't capitalize "<Resource> Access".

File: docs/pages/access-controls/device-trust/guide.mdx [hunk 3]
```diff
@@ -4,14 +4,14 @@ description: Get started with Teleport Device Trust
...
 ---
 
-<Admonition type="warning">
+<Admonition type="notice">
   Device Trust supports the following components:
 
   - User devices: macOS, Windows and Linux.
   - Teleport client: `tsh` and Teleport connect.
-  - Resources: SSH, Database and Kubernetes.
+  - Resources: Apps (role-based only), SSH, Database and Kubernetes
```
Comment (ptgott) [inline docs/pages/access-controls/device-trust/guide.mdx:12]:
```suggestion
  - Resources: apps (role-based only), SSH nodes, databases, and Kubernetes clusters.
```
To make this a list of resources

Comment (ptgott) [reply docs/pages/access-controls/device-trust/guide.mdx:12]: Also, I'm not sure the docs talk about what a role-based app is elsewhere—is there another wording we can use for this?

File: docs/pages/access-controls/device-trust/guide.mdx [hunk 4]
```diff
@@ -4,14 +4,14 @@ description: Get started with Teleport Device Trust
...
 
-  Support for Web UI and application access are planned for upcoming Teleport
+  Support for Web UI and Desktop Access are planned for upcoming Teleport
```
Comment (ptgott) [inline docs/pages/access-controls/device-trust/guide.mdx:14]:
```suggestion
  Support for the Web UI and desktop access are planned for upcoming Teleport
```

File: docs/pages/includes/device-trust/troubleshooting.mdx [hunk 5]
```diff
@@ -23,3 +23,26 @@ group and assign it to your OS user. If that is not possible, or you are looking
 for a different solution, we recommend creating udev rules similar to the ones
 shipped by the [TPM2 Software Stack](
 https://github.com/tpm2-software/tpm2-tss/blob/ede63dd1ac1f0a46029d457304edcac2162bfab8/dist/tpm-udev.rules#L4).
+
+### App Access and "access to this app requires a trusted device"
```
Comment (ptgott) [inline docs/pages/includes/device-trust/troubleshooting.mdx:27]:
```suggestion
### App access and "access to this app requires a trusted device"
```

File: docs/pages/includes/device-trust/troubleshooting.mdx [hunk 6]
```diff
@@ -23,3 +23,26 @@ group and assign it to your OS user. If that is not possible, or you are looking
...
+
+A Teleport admin configured the particular app you are trying to access to
+require a trusted device.
+
+There are a few situations that might cause the error:
+
+1. You are trying to access the app using the Web UI
+
+2. You are trying to access the app using an untrusted device
```
Comment (ptgott) [inline docs/pages/includes/device-trust/troubleshooting.mdx:36]:
```suggestion
1. You are trying to access the app using the Web UI.

2. You are trying to access the app using an untrusted device.
```

**Review by ptgott — APPROVED** [review]

File: docs/pages/access-controls/device-trust/guide.mdx (hunk 3 above)
Comment (zmb3) [reply docs/pages/access-controls/device-trust/guide.mdx:12]: How about "role-based enforcement only" ??

Comment (codingllama) [reply docs/pages/access-controls/device-trust/guide.mdx]: Applied in both instances, used "role-based enforcement only".

**Comment by codingllama** [comment]
Thanks everyone!

**Pull request merged**